### PR TITLE
[PW-1028] ACH on the frontend and backend

### PR DESCRIPTION
**Pull request by cyattilakiss**
**Description**
Add ACH checkout support 
- add bank account number
 - add bank location id
 - add bank account owner name
 - add validation on frontend

### CODE REVIEW

File: Observer/AdyenHppDataAssignObserver.php [hunk 1]
```diff
@@ -39,7 +39,10 @@ class AdyenHppDataAssignObserver extends AbstractDataAssignObserver
     const DF_VALUE = 'df_value';
     const SSN = 'ssn';
 	const OWNERNAME = 'ownerName';
+    const BANKACCOUNTOWNERNAME = 'bankAccountOwnerName';
```
Comment (rikterbeek) [inline Observer/AdyenHppDataAssignObserver.php]: Can you add the underscore to make it more readable for all of these constants ? So BANKACCOUNT_OWNER_NAME BANKACCOUNT_NUMBER, BANK_LOCATION_ID.

File: Observer/AdyenHppDataAssignObserver.php [hunk 2]
```diff
@@ -54,7 +57,10 @@ class AdyenHppDataAssignObserver extends AbstractDataAssignObserver
         self::DF_VALUE,
         self::SSN,
 		self::OWNERNAME,
-		self::IBANNUMBER
+        self::BANKACCOUNTOWNERNAME,
+		self::IBANNUMBER,
+        self::BANKACCOUNTNUMBER,
+        self::BANKLOCATIONID
```
Comment (rikterbeek) [inline Observer/AdyenHppDataAssignObserver.php]: same here

File: view/frontend/web/template/payment/hpp-form.html [hunk 3]
```diff
@@ -221,6 +221,72 @@
...
+                                                        minlength: 9,
+                                                        maxlength: 9
+                                                    },
+                                                    value: bankLocationId"
+                                       data-validate="{required:true}"
+                                />
+                            </div>
+                        </div>
+
+                    <!--/ko-->
```
Comment (rikterbeek) [inline view/frontend/web/template/payment/hpp-form.html]: did you checked with checkout regarding component ?

File: Observer/AdyenHppDataAssignObserver.php (hunk 1 above)
Comment (cyattilakiss) [reply Observer/AdyenHppDataAssignObserver.php]: indeed

**Pull request merged**